app/web/components/CustomColorSwitch.tsx

```diff
@@ -0,0 +1,84 @@
+import { Switch, withStyles } from "@material-ui/core";
+import makeStyles from "utils/makeStyles";
+
+import CircularProgress from "./CircularProgress";
+
+const useStyles = makeStyles(({ palette, shadows }) => ({
+  circle: {
+    display: "flex",
+    alignItems: "center",
+    justifyContent: "center",
+    width: 20,
+    height: 20,
+    borderRadius: "50%",
+    backgroundColor: palette.grey[600],
+    boxShadow: shadows[1],
+  },
+  active: {
+    backgroundColor: palette.grey[600],
+  },
+}));
+
+export default function CustomColorSwitch({
+  checked,
+  color,
+  isLoading,
+  onClick,
+  status,
+}: {
+  checked: boolean;
+  color: string;
+  isLoading: boolean;
+  onClick: () => void;
+  status: string;
+}) {
+  const classes = useStyles();
+
+  const CustomColorSwitch = withStyles((theme) => ({
+    switchBase: {
+      color: theme.palette.grey[600],
+      "& + $track": {
+        backgroundColor: theme.palette.grey[200],
+      },
+      "&$checked": {
+        color,
+        "& + $track": {
+          backgroundColor: color,
+        },
+      },
+      "&$disabled": {
+        color: checked ? color : theme.palette.grey[600],
+        "& + $track": {
+          backgroundColor: checked ? color : theme.palette.grey[200],
+          opacity: 0.4,
+        },
+      },
+    },
+    checked: {},
+    disabled: {},
+    track: {},
+  }))(Switch);
+
+  const Icon = () => (
+    <div
+      className={`${classes.circle} ${
+        checked && !isLoading ? classes.active : ""
+      }`}
```

Comment:
I think there's a helper for this from `import classNames from "classnames";`.

I think this would be `classNames(classes.circle, {classes.active: checked && !isLoading})`

Reply:
Okay this is done!